test/dynamo/test_repros.py

```diff
@@ -2226,6 +2227,79 @@ def f(x):
         )
         self.assertEqual(gm(inp).shape, f(inp).shape)
 
+    @patch.object(torch._dynamo.config, "dynamic_shapes", True)
+    def test_dynamic_slicing(self):
+        def f(x):
+            return x[: x.shape[0] - 2, x.shape[1] - 1 :: 2]
+
+        gm_aten_mode, _ = torch._dynamo.export(
+            f, torch.randn(4, 5), aten_graph=True, tracing_mode="symbolic"
+        )
+
+        inp = torch.randn(6, 7)
+        self.assertEqual(gm_aten_mode(inp).shape, f(inp).shape)
+
+        count = 0
+        for node in gm_aten_mode.graph.nodes:
+            if (
+                node.op == "call_function"
+                and node.target == torch.ops.aten.slice.Tensor
+            ):
+                count += 1
+
+        self.assertEqual(count, 2)
+
+        gm_dynamo_mode, _ = torch._dynamo.export(f, torch.randn(4, 5), aten_graph=False)
+
+        # at this point, the graph should contain 3 getitem methods
+        # one for x.shape[0]-2 and one for x.shape[1]-1 and one for slice
+        count = 0
+        for node in gm_dynamo_mode.graph.nodes:
+            if node.op == "call_function" and node.target == operator.getitem:
+                count += 1
```

Comment:
The logic differs a bit, because the representation of lookup differs - but it makes this test confusing. Add a comment?